From 072380b1a1391b2825021deddd8e88baa234725d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <cynerd@email.cz>
Date: Fri, 17 Apr 2020 09:58:10 +0200
Subject: [PATCH] tools: add autoconf-archive
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Some autotools based builds systems are using autoconf-archive scripts
and are expected to almost always be available. It is not always
possible to get release tar ball and so sometimes for autoreconf this is
needed.

Including this adds little to no overhead in terms of build time as
those are just m4 scripts copied to appropriate location.

Signed-off-by: Karel Kočí <karel.koci@nic.cz>
---
 tools/Makefile                  |  2 +-
 tools/autoconf-archive/Makefile | 34 +++++++++++++++++++++++++++++++++
 2 files changed, 35 insertions(+), 1 deletion(-)
 create mode 100644 tools/autoconf-archive/Makefile

diff --git a/tools/Makefile b/tools/Makefile
index d7207ba..24eb177 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -22,7 +22,7 @@ ifneq ($(CONFIG_SDK)$(CONFIG_PACKAGE_kmod-b43)$(CONFIG_PACKAGE_b43legacy-firmwar
 endif
 
 tools-$(BUILD_TOOLCHAIN) += gmp mpfr mpc libelf expat
-tools-y += m4 libtool autoconf automake flex bison pkg-config mklibs zlib
+tools-y += m4 libtool autoconf autoconf-archive automake flex bison pkg-config mklibs zlib
 tools-y += sstrip make-ext4fs e2fsprogs mtd-utils mkimage
 tools-y += firmware-utils patch-image quilt padjffs2
 tools-y += mm-macros missing-macros cmake scons bc findutils gengetopt patchelf
diff --git a/tools/autoconf-archive/Makefile b/tools/autoconf-archive/Makefile
new file mode 100644
index 0000000..74415b3
--- /dev/null
+++ b/tools/autoconf-archive/Makefile
@@ -0,0 +1,34 @@
+# 
+# Copyright (C) 2006-2020 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=autoconf-archive
+PKG_VERSION:=2019.01.06
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
+PKG_SOURCE_URL:=@GNU/autoconf-archive
+PKG_HASH:=17195c833098da79de5778ee90948f4c5d90ed1a0cf8391b4ab348e2ec511e3f
+
+include $(INCLUDE_DIR)/host-build.mk
+
+HOST_CONFIGURE_ARGS += \
+	--datarootdir=$(STAGING_DIR_HOST)/share
+
+define Host/Compile
+	export SHELL="$(BASH)"; $(MAKE) -C $(HOST_BUILD_DIR)
+endef
+
+define Host/Install
+	export SHELL="$(BASH)"; $(MAKE) -C $(HOST_BUILD_DIR) install
+endef
+
+define Host/Clean
+	-export SHELL="$(BASH)"; $(MAKE) -C $(HOST_BUILD_DIR) uninstall
+	$(call Host/Clean/Default)
+endef
+
+$(eval $(call HostBuild))
-- 
2.26.0

